### PR TITLE
Flow integration

### PR DESCRIPTION
**Pull request by felipecalvo**
:warning: **We need to tag and release flow first**

This PR adds the necessary migrations, alias and handy methods for mumukit-flow integration.

It also loses some navigation modules, which are moved to flow: https://github.com/mumuki/mumukit-flow/pull/11

Once we have a version number for flow, it should be removed from Gemfile and added to the gemspec file.

### CODE REVIEW

File: app/models/exercise.rb [hunk 1]
```diff
@@ -36,9 +38,14 @@ class Exercise < ApplicationRecord
   validates_presence_of :submissions_count,
                         :guide, :bibliotheca_id
 
+  alias_attribute :parent, :guide
```
Comment (luchotc) [inline app/models/exercise.rb]: We are removing the concept of both parent and children in favor of structural_* and navigable_*.
In this case, we should use structural_parent which is already defined.

File: app/models/guide.rb [hunk 2]
```diff
@@ -8,11 +8,15 @@ class Guide < Content
...
           WithLanguage
 
+  include Mumukit::Flow::AdaptiveItem
+
   markdown_on :corollary, :sources, :learn_more, :teacher_info
 
   numbered :exercises
   has_many :exercises, -> { order(number: :asc) }, dependent: :destroy
 
+  alias_attribute :children, :exercises
```
Comment (luchotc) [inline app/models/guide.rb]: Idem for this one

File: app/models/guide.rb [hunk 3]
```diff
@@ -123,6 +127,10 @@ def resettable?
     usage_in_organization.resettable?
   end
 
+  def exercise_assignments_for(user)
```
Comment (luchotc) [inline app/models/guide.rb]: This one seems a bit odd here, but I know it's required by flow and with the current structure there isn't a better place to implement it, so :+1:

File: app/models/exercise.rb (hunk 1 above)
Comment (felipecalvo) [reply app/models/exercise.rb]: So I should delete this alias (and the other one) and just use structural_parent/structural_children in flow?

File: app/models/guide.rb (hunk 3 above)
Comment (felipecalvo) [reply app/models/guide.rb]: ┐(︶▽︶)┌

**Comment by luchotc** [comment]
https://rubygems.org/gems/mumukit-flow :)

**Comment by felipecalvo** [comment]
This is not happening soon and might have to be re-done anyway.

**Pull request closed without merging**